server/src/main/java/com/tinyfs/config/CredentialsConfig.java

```diff
@@ -1,27 +1,50 @@
 package com.tinyfs.config;
 
+import java.io.File;
+import java.util.Arrays;
 import java.util.Base64;
+import java.util.Scanner;
 
 import javax.crypto.SecretKey;
 import javax.crypto.spec.SecretKeySpec;
 
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import com.google.common.base.Throwables;
+import com.tinyfs.exception.CredentialsException;
+
 @Configuration
 public class CredentialsConfig {
 
-  /**
-   * This key is in no way secure, and merely to serve as a proof of concept.
-   * This project will use a much more secure scheme, where a key is not statically used from code.
-   *
-   * @TODO Pull a key from a local file in the repository, not importing it statically. RemoteTinyFS/issues/2
-   */
-  private static final String DES_KEY_RAW = "a9/Vbr+oosQ=";
-  private static final byte[] DES_KEY_DECODED = Base64.getDecoder().decode(DES_KEY_RAW);
+  public static final String USERNAME_LIST_KEY = "usernames";
+  public static final String CREDENTIALS_ENCRYPTION_SCHEME = "AES";
+
+  private static final String CREDENTIALS_KEY_NAME = "credentials.key";
+  private static final int AES_KEY_LENGTH = 16;
 
   @Bean
-  public SecretKey secretKey() {
-    return new SecretKeySpec(DES_KEY_DECODED, 0, DES_KEY_DECODED.length, "DES");
+  public SecretKey secretKey() throws CredentialsException {
+    byte[] bytes = Arrays.copyOf(
+      Base64.getDecoder().decode(getKeyText(CREDENTIALS_KEY_NAME)),
+      AES_KEY_LENGTH);
+    return new SecretKeySpec(
+      bytes,
+      0,
+      AES_KEY_LENGTH,
+      CREDENTIALS_ENCRYPTION_SCHEME);
+  }
+
+  private String getKeyText(final String resourceName) throws CredentialsException {
+    File file = new File(CredentialsConfig.class.getClassLoader().getResource("credentials.key").getFile());
```

Comment:
I'd be fine making the server generate it if it doesn't have one, but I'd rather do it in a separate PR.